### DISCUSSION TITLE
Show play stats

### DISCUSSION THREAD

**Discussion by robbevp** [accentor/web#644] [Ideas]
Since we track plays - we should also provide the user some stats about their listening habits:

Some first ideas:
Select time periods:
* last 7 days
* last month
* last 12 months
* current calendar year
* previous calendar year
* all time
* custom range

Show stats both for the amount of plays and the amount of time listened (so play count * track length)

Show following stats:
* top x artists
* top x tracks
* top x albums

Some other stats that could be interesting:
* Total time spent listening
* % of library that has been played (in # of tracks)

Other ideas?

**Comment by chvp** [comment]
What time of day the user listens to music (some sort of heatmap or punchcard of plays).

**Comment by robbevp** [comment]
Another idea would be something like favourite song from an album you don't like or song you don't like from an album you like (where we find the songs where the play count differs the most (either positive or negative) from the average of the album they are on
